docs/4.x/tests-github.md

```diff
@@ -0,0 +1,50 @@
+---
+category: Develop
+previous: tests-js-tracker
+next: tests-troubleshooting
+---
+# Running tests/builds on GitHub Action
+
+[Github Action](https://github.com/features/actions) makes it easy to automate all your software workflows, now with world-class CI/CD. Build, test, and deploy your code right from GitHub. Make code reviews, branch management, and issue triaging work the way you want.
+
+## Matomo on GitHub Action
+
+Matomo uses GitHub Action to automatically run its build by various triggers.
+- [Build Tracker JS] Trigger by the comment `build js` into the pull request. That will compress js/piwik.js into matomo.js and piwik.js
+- [Build VUE] Auto trigger, it will build VUE file and commit back to the PR
+- [Composer Update] Auto trigger by cron，it will run `composer update` and commit back to the repo
+- [PHPCS check] Auto trigger, checking PHPCS code quality if not valid will return error details.
+- [Inactive PR] after 14 days, inactive PR will mark as state
+
+
+Each developer is responsible to keep the commits/pull requests build green.
+
+## GitHub Action Scripts
+
+All the GitHub Action build file are locate in `.github/workflows/*.yml`
+
+Each script is triggered by its own conditions, for more details see [Github Action Docs](https://docs.github.com/en/actions)
+
+## Create a new CI script
+
+To create a new CI script please read [Quickstart for GitHub Actions](https://docs.github.com/en/actions/quickstart)
+
+We recommend forking the Matomo project and running actions on your own pipeline first. Also, please check the following list before getting started:
+
+### Using other action
+If you need to use an action from the marketplace, please ensure that it is either an official action provided by GitHub, or you did a review of the actions code. In latter case, please ensure to use a version fixed by a full length commit SHA.
+
+### Security
+There is a chance the action scripts getting hacked, there are two common attack [script injection attacks](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#example-of-a-script-injection-attack) and [Security hardening](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions).
+
+For more details, refer [this article](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#overview)
```

Comment:
can we mention that paying attention to security is of the upmost importance as for example other actions can steal the GitHub token and get up to write access to our code https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#stealing-the-jobs-github_token